src/Platform.Handlers/tests/Xamarin.Platform.Handlers.DeviceTests/Handlers/HandlerTestBase.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Essentials;
+using Xunit;
+
+namespace Xamarin.Platform.Handlers.DeviceTests
+{
+	public partial class HandlerTestBase<THandler> : TestBase
+		where THandler : IViewHandler
+	{
+
+		public Task<T> InvokeOnMainThreadAsync<T>(Func<T> func) =>
+			MainThread.InvokeOnMainThreadAsync(func);
+
+		public Task InvokeOnMainThreadAsync(Action action) =>
+			MainThread.InvokeOnMainThreadAsync(action);
+
+		protected async Task<THandler> CreateHandlerAsync(IView view)
```

Comment:
iOS has to run with all of this marshalled over to the UI Thread otherwise it throws consistency exceptions.